src/network/networkd-route.c

```diff
@@ -2356,6 +2376,67 @@ int config_parse_multipath_route(
         return 0;
 }
 
+int config_parse_route_table_names(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_free_ char *name = NULL;
+        Hashmap **s = data;
+        uint32_t table;
+        const char *p;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                *s = hashmap_free_free_key(*s);
+                return 0;
+        }
+
+        p = rvalue;
+        r = extract_first_word(&p, &name, ":", 0);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r <= 0 || isempty(p)) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Invalid RouteTable=, ignoring assignment: %s", rvalue);
+                return 0;
+        }
+
+        r = safe_atou32(p, &table);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse RouteTable=, ignoring assignment: %s", p);
+                return 0;
+        }
+
+        if (table == 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Invalid RouteTable=, ignoring assignment: %s", p);
+                return 0;
+        }
+
```

Comment:
Maybe, drop the case that `name` is `"default"`, `"main"`, or `"local"`.

Reply:
It's ok if user wants them there.

Reply:
Hmm, but in `route_table_from_string_full()` we first check the string is one of `default`, `main` or `local`. So, setting `RouteTable=default:12345` is completely ignored without any warning...

Reply:
We will just give a warning log

Reply:
like 
```
etc/systemd/networkd.conf:3: Route table name local alreay preconfigured. Ignoring assignment: local:255
/etc/systemd/networkd.conf:4: Route table name main alreay preconfigured. Ignoring assignment: main:254
/etc/systemd/networkd.conf:5: Route table name default alreay preconfigured. Ignoring assignment: default:253
```